source/grid/grid_out.cc

```diff
@@ -1066,10 +1064,11 @@ void GridOut::write_msh (const Triangulation<dim, spacedim> &tria,
 
   // write faces and lines with
   // non-zero boundary indicator
+  unsigned int next_index = tria.n_active_cells()+1;
   if (msh_flags.write_faces)
-    write_msh_faces (tria, cell_index, out);
+    write_msh_faces (tria, next_index, out);
   if (msh_flags.write_lines)
-    write_msh_lines (tria, cell_index, out);
+    write_msh_lines (tria, next_index, out);
```

Comment:
I am confused what happens here (also in the original code). We give unique ids to cells and faces because we keep counting upwards with next_index, but faces and lines would share the same index? I wonder if writing faces and lines will produce a broken file. Or do we never output faces and lines at the same time? Either way, I think write_msh_faces should return/modify the new next_index.

Btw., it took me a while to see what is going on (continue counting after cells). I think the old code was more clear in that sense. Not a big issue of course.